org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProvider.java

```diff
@@ -171,7 +173,24 @@ public void fixEmptyEnumLiteral(final Issue issue, IssueResolutionAcceptor accep
 					enumLiteralDeclaration.setLiteral(keyword);
 				});
 	}
-
+	
+	@Fix(SPACES_IN_KEYWORD)
+	public void fixKeywordNoSpaces(final Issue issue, IssueResolutionAcceptor acceptor) {
+		acceptor.accept(issue, "Split keyword with spaces", "Split keyword with spaces", NULL_QUICKFIX_IMAGE,
+				(ISemanticModification) (EObject element, IModificationContext context) -> {
+					final Keyword keyword = (Keyword) element;
+					final String[] identifiers = keyword.getValue().trim().split("\\s+");
+					if (identifiers.length > 0) {
+						final IXtextDocument document = context.getXtextDocument();
+						final String quote = String.valueOf(document.getChar(issue.getOffset()));
+						document.replace(issue.getOffset(), issue.getLength(), Arrays.stream(identifiers)
+								.map(identifier -> String.format("%s%s%s", quote, identifier, quote)).collect(Collectors.joining(" ")));
```

Comment:
I know it's a matter of taste, but I find `quote + Joiner.on(quote + ' ' + quote).join(identifiers) + quote` easier to read.